src/wp-admin/includes/class-wp-debug-data.php

```diff
@@ -1223,6 +1038,196 @@ public static function debug_data() {
+				);
+			} else {
+				$fields['memory_limit'] = array(
+					'label' => __( 'PHP memory limit' ),
+					'value' => ini_get( 'memory_limit' ),
+				);
+			}
+
+			$fields['max_input_time']      = array(
+				'label' => __( 'Max input time' ),
+				'value' => ini_get( 'max_input_time' ),
+			);
+			$fields['upload_max_filesize'] = array(
+				'label' => __( 'Upload max filesize' ),
+				'value' => ini_get( 'upload_max_filesize' ),
+			);
+			$fields['php_post_max_size']   = array(
+				'label' => __( 'PHP post max size' ),
+				'value' => ini_get( 'post_max_size' ),
+			);
+		}
+
+		if ( function_exists( 'curl_version' ) ) {
+			$curl = curl_version();
+
+			$fields['curl_version'] = array(
+				'label' => __( 'cURL version' ),
+				'value' => sprintf( '%s %s', $curl['version'], $curl['ssl_version'] ),
+			);
+		} else {
+			$fields['curl_version'] = array(
+				'label' => __( 'cURL version' ),
+				'value' => __( 'Not available' ),
+				'debug' => 'not available',
+			);
+		}
+
+		// SUHOSIN.
+		$suhosin_loaded = ( extension_loaded( 'suhosin' ) || ( defined( 'SUHOSIN_PATCH' ) && constant( 'SUHOSIN_PATCH' ) ) );
+
+		$fields['suhosin'] = array(
+			'label' => __( 'Is SUHOSIN installed?' ),
+			'value' => ( $suhosin_loaded ? __( 'Yes' ) : __( 'No' ) ),
+			'debug' => $suhosin_loaded,
+		);
+
+		// Imagick.
+		$imagick_loaded = extension_loaded( 'imagick' );
+
+		$fields['imagick_availability'] = array(
+			'label' => __( 'Is the Imagick library available?' ),
+			'value' => ( $imagick_loaded ? __( 'Yes' ) : __( 'No' ) ),
+			'debug' => $imagick_loaded,
+		);
+
+		// Pretty permalinks.
+		$pretty_permalinks_supported = got_url_rewrite();
+
+		$fields['pretty_permalinks'] = array(
+			'label' => __( 'Are pretty permalinks supported?' ),
+			'value' => ( $pretty_permalinks_supported ? __( 'Yes' ) : __( 'No' ) ),
+			'debug' => $pretty_permalinks_supported,
+		);
+
+		// Check if a .htaccess file exists.
+		if ( is_file( ABSPATH . '.htaccess' ) ) {
+			// If the file exists, grab the content of it.
+			$htaccess_content = file_get_contents( ABSPATH . '.htaccess' );
+
+			// Filter away the core WordPress rules.
+			$filtered_htaccess_content = trim( preg_replace( '/\# BEGIN WordPress[\s\S]+?# END WordPress/si', '', $htaccess_content ) );
+			$filtered_htaccess_content = ! empty( $filtered_htaccess_content );
+
+			if ( $filtered_htaccess_content ) {
+				/* translators: %s: .htaccess */
+				$htaccess_rules_string = sprintf( __( 'Custom rules have been added to your %s file.' ), '.htaccess' );
+			} else {
+				/* translators: %s: .htaccess */
+				$htaccess_rules_string = sprintf( __( 'Your %s file contains only core WordPress features.' ), '.htaccess' );
+			}
+
+			$fields['htaccess_extra_rules'] = array(
+				'label' => __( '.htaccess rules' ),
+				'value' => $htaccess_rules_string,
+				'debug' => $filtered_htaccess_content,
+			);
+		}
+
+		// Server time.
+		$date = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
```

Comment:
@dmsnell PHPstorm warned me that this line could possibly throw an exception. Since we have a highly controlled set of parameters, I think it's safe to assume we will never get this exception.

Yet PHPstorm wanted me to add a `@throws` in the PHPdoc, which I ignored for the time being. But what do you suggest what we do about it?

Adding a `@throws` will just bubble up the problem. 

How about
```suggestion
		try {
			$date = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
		} catch ( Exception $e ) {
			// It should be impossible to reach this statement. But in case let's fail gracefully.
			wp_die( $e->get_message() );
		}
```

(Untested and written in GitHub only)

Reply:
While not impossible (See the [Europe/Kyiv change](https://mm.icann.org/pipermail/tz-announce/2022-August/000071.html) in the "Changes to zone name" section - Which changed for PHP 8.2), since we're using the UTC timezone, the chance of an Exception being thrown for this call is extremely low.

It seems to be fairly safe to consider this a tooling flag that can be ignored.

I don't have strong feelings on this though. It'll probably just come down to preference.

Reply:
I don't see any notes in the documentation suggesting that `UTC` was ever _not_ part of the distribution. The constant `DateTimeZone::UTC` hints at this, and if you ask PHP to get a timezone supported by UTC it returns the string in use here.

```php
php > var_dump( DateTimeZone::listIdentifiers( DateTimeZone::UTC ) );
array(1) {
  [0]=>
  string(3) "UTC"
}
```

While sometimes the documentation is wrong, it seems unlikely here, meaning that it may not even be an extremely low chance of throwing here, but not possible.

Do we need a WPCS ignore statement here? If not then I think it's fine as-is. Regardless, as I have written in other comments - this PR is about modularizing a piece of the debug data code, and I recommend against making semantic changes otherwise until that factoring has completed.

Reply:
It does not trigger WPCS, just triggers the Code analysis from PHPstorm.

So we can keep it as is.